release/ray_release/cluster_manager/minimal.py

```diff
@@ -248,6 +251,40 @@ def create_cluster_compute(self, _repeat: bool = True):
                     f"ID {self.cluster_compute_id}"
                 )
 
+    def annotate_cluster_compute(
+        self,
+        cluster_compute: Dict[str, Any],
+        cloud_provider: str,
+        extra_tags: Dict[str, str],
+    ) -> Dict[str, Any]:
+        if not extra_tags or cloud_provider != "AWS":
+            return cluster_compute
+
+        cluster_compute = cluster_compute.copy()
+        aws = deepcopy(cluster_compute.get("aws", {}))
+        cluster_compute["aws"] = aws
+        tag_specifications = aws.setdefault("TagSpecifications", [])
+        for resource in RELEASE_AWS_RESOURCE_TYPES_TO_TRACK_FOR_BILLING:
+            resource_tags: dict = next(
+                (
+                    x
+                    for x in tag_specifications
+                    if x.get("ResourceType", "") == resource
+                ),
+                None,
+            )
+            if resource_tags is None:
+                resource_tags = {"ResourceType": resource, "Tags": []}
+                tag_specifications.append(resource_tags)
+            tags = resource_tags["Tags"]
+            for key, value in extra_tags.items():
+                tags.append({"Key": key, "Value": value})
+        return cluster_compute
```

Comment:
Let's add some comments here

```suggestion
        for resource in RELEASE_AWS_RESOURCE_TYPES_TO_TRACK_FOR_BILLING:
            # Check if there is already a tag specification for the resource.
            # If so, return first item.
            resource_tags: dict = next(
                (
                    spec
                    for spec in tag_specifications
                    if spec.get("ResourceType", "") == resource
                ),
                None,
            )
            
            # If no tag specification exists, add
            if resource_tags is None:
                resource_tags = {"ResourceType": resource, "Tags": []}
                tag_specifications.append(resource_tags)
                
            # Add our tags to the specification
            tags = resource_tags["Tags"]
            for key, value in extra_tags.items():
                tags.append({"Key": key, "Value": value})
        return cluster_compute
```